archiver.go

```diff
@@ -132,6 +133,19 @@ func (a *Archiver) do(log log15.Logger, j *Job) (err error) {
 		"last-fetch", r.FetchedAt,
 		"references", len(r.References))
 
+	defer func() {
+		if rcv := recover(); rcv != nil {
+			log.Error("panic while processing repository", "error", rcv)
+
+			r.FetchErrorAt = &now
+			if sErr := a.Store.UpdateFailed(r, model.Pending); sErr != nil {
+				log15.Error("error setting repo as failed", "id", r.ID, "err", err)
+			}
+
+			err = fmt.Errorf("%v", goerrors.Wrap(rcv, 3).ErrorStack())
```

Comment:
Actually, I don't think the `goerrors` is needed at all. https://play.golang.org/p/wVj32VTnYpH

`debug.Stack()` would work fine (only for panics)

Reply:
Ok, changing it. I just looked for `goerrors` because it allow you  to prune those stacktraces coming from all the recover stuff, but about adding dependencies the less the better